src/project-manager.ts

```diff
@@ -761,14 +761,18 @@ export class InMemoryFileSystem implements ts.ParseConfigHost, ts.ModuleResoluti
 	 * @param path file path (both absolute or relative file paths are accepted)
 	 */
 	fileExists(path: string): boolean {
-		return this.readFile(path) !== undefined;
+		return this.readFileImpl(path) !== undefined;
 	}
 
 	/**
 	 * @param path file path (both absolute or relative file paths are accepted)
-	 * @return file's content in the following order (overlay then cache) if any
+	 * @return file's content in the following order (overlay then cache)
 	 */
-	readFile(path: string): string | undefined {
+	readFile(path: string): string {
+		return this.readFileImpl(path) || '';
```

Comment:
Won't work, sorry. There might be a case when TS compiler requests file that does not exists and if we'll throw an error it may break source analysis